lumigator/python/mzai/backend/backend/tests/conftest.py

```diff
@@ -107,40 +111,79 @@ def db_session(db_engine: Engine):
+#         yield localstack
 
 
-@pytest.fixture(scope="session", autouse=True)
-def setup_aws(localstack_container: LocalStackContainer):
-    """Setup env vars/AWS resources for use with the LocalStack container."""
+@pytest.fixture(scope="function")
+def fake_s3_storage() -> dict:
+    return dict()
+
+@pytest.fixture(scope="function")
+def fake_s3_client(fake_s3_storage) -> S3Client:
+    os.environ["AWS_ACCESS_KEY_ID"] = "test"
+    # Please check https://github.com/localstack/localstack/issues/5894
+    # for info about the test region used
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "test"  # pragma: allowlist secret
+    os.environ["AWS_ENDPOINT_URL"] = "http://example.com:4566"
+    return FakeS3Client(fake_s3_storage)
+
+@pytest.fixture(scope="function")
+def boto_s3_client(fake_s3_storage) -> S3Client:
     # Initialize S3
-    s3 = localstack_container.get_client("s3")
-    try:
-        s3.create_bucket(
-            Bucket=settings.S3_BUCKET,
-            CreateBucketConfiguration={"LocationConstraint": localstack_container.region_name},
-        )
-    except ClientError as e:
-        if e.response["Error"]["Code"] == "BucketAlreadyOwnedByYou":
-            print("Bucket already created, continue")
-        else:
-            raise e
-    # add ENV vars for FSSPEC access to S3 (s3fs + HuggingFace datasets)
-    os.environ["FSSPEC_S3_KEY"] = "testcontainers-localstack"
-    os.environ["FSSPEC_S3_SECRET"] = "testcontainers-localstack"  # pragma: allowlist secret
-    os.environ["FSSPEC_S3_ENDPOINT_URL"] = localstack_container.get_url()
-    os.environ["AWS_ACCESS_KEY_ID"] = "testcontainers-localstack"
-    os.environ["AWS_SECRET_ACCESS_KEY"] = "testcontainers-localstack"  # pragma: allowlist secret
-    os.environ["AWS_ENDPOINT_URL"] = localstack_container.get_url()
+    os.environ["AWS_ACCESS_KEY_ID"] = "test"
+    # Please check https://github.com/localstack/localstack/issues/5894
+    # for info about the test region used
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "test"  # pragma: allowlist secret
+    os.environ["AWS_ENDPOINT_URL"] = "http://localhost:4566"
+    return boto3.client("s3")
+
+# @pytest.fixture(scope="function")
+# def localstack_s3_client(localstack_container: LocalStackContainer) -> S3Client:
+#     client = localstack_container.get_client("s3")
+#     mock_client = Mock(wraps=client)
+#     yield mock_client
+#     print(f'---> s3client: {str(mock_client.mock_calls)}')
+
+@pytest.fixture(scope="function")
+def fake_s3fs(fake_s3_storage) -> S3FileSystem:
+    # ...and patch the s3fs name with the Fake
+    fsspec.register_implementation("s3", MemoryFileSystem, clobber=True, errtxt="Failed to register mock S3FS")
+    # fsspec.register_implementation("s3", FakeS3FS, clobber=True, errtxt="Failed to register mock S3FS")
+    # FakeS3FS.storage = fake_s3_storage
```

Comment:
remove comments?

Reply:
Totally. I left them just in case the memfs didn't work, since it seems to work ok, I'll just remove it.